### PR TITLE
Extra utilities

### PR DESCRIPTION
**Pull request by mateoi**
These are utilities used in the cubes affiliated package, but that have wider uses than only in cubes.
They are mostly WCS-related, but I also included a scipy implementation of the (very awesome) Savitzky-Golay filter for smoothing noisy data.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,7 +34,6 @@ sunpy/data/sample/*.fit
 
 ### https://raw.github.com/github/gitignore/master/Global/OSX.gitignore
 
-.DS_Store
```
Comment (dpshelio) [inline .gitignore]: why is this line removed? is not that the annoying folder from OSx?

Comment (wafels) [inline .gitignore]: Why not just import the SciPy implementation of the Savitzky-Golay filter
instead of using the scipy cookbook version?

http://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.savgol_filter.html#scipy.signal.savgol_filter

Comment (dpshelio) [inline .gitignore]: Forget it! you are right! it's on the first line :)

Comment (mateoi) [inline .gitignore]: @dpshelio - .DS_Store is already in.
@wafels - you're right, I didn't think about checking!

File: sunpy/util/util.py [hunk 2]
```diff
@@ -9,8 +9,9 @@
 
 __all__ = ['to_signed', 'unique', 'print_table',
            'replacement_filename', 'goes_flare_class', 'merge', 'common_base',
-           'minimal_pairs', 'polyfun_at',
-           'expand_list', 'expand_list_generator', 'Deprecated']
+           'minimal_pairs', 'polyfun_at', 'expand_list',
+           'expand_list_generator', 'Deprecated', 'savitzky_golay']
```
Comment (ayshih) [inline sunpy/util/util.py]: Remove here too

File: sunpy/util/util.py [hunk 3]
```diff
@@ -179,12 +186,13 @@ def replacement_filename(path):
                 return newpath
 
 
-#==============================================================================
+# =============================================================================
 # expand list from :http://stackoverflow.com/a/2185971/2486799
-#==============================================================================
+# =============================================================================
 def expand_list(input):
```
Comment (wafels) [inline sunpy/util/util.py]: Does this function work on, or need to work on, arbitrarily nested lists?  Googling around, it seems there are many possible implementations of this functionality.  Also, where is it used in sunpy or in this PR?  I couldn't find it used anywhere.

Comment (mateoi) [inline sunpy/util/util.py]: No idea - I didn't add that. I agree with you that we wouldn't need to use this.

Comment (ayshih) [inline sunpy/util/util.py]: > Also, where is it used in sunpy or in this PR? I couldn't find it used anywhere.

In `Map` stuff`: https://github.com/sunpy/sunpy/search?utf8=%E2%9C%93&q=expand_list

Comment (dpshelio) [inline sunpy/util/util.py]: it was @rhewett in 7f13697d

**Comment by dpshelio** [comment]
@mateoi could you rebase it? Thanks :)
By the way, it would have been a lot easier to review if the "style" corrections of files that it has not to do with this PR would have been separated in a different one...

@sunpy/sunpy-developers these wcs tools are needed for [sunpy/cube](https://github.com/sunpy/cube) and [MSSLSolar/EISpy](https://github.com/MSSLSolar/EISpy) to work. Review please :)

File: sunpy/wcs/wcs_util.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+
+def reindex_wcs(wcs, inds):
+    # From astropy.spectral_cube.wcs_utils
+    """
+    Re-index a WCS given indices.  The number of axes may be reduced.
+
+    Parameters
+    ----------
+    wcs: sunpy.wcs.wcs.WCS
```
Comment (Cadair) [inline sunpy/wcs/wcs_util.py]: type should have backticks around it so that it is lined to by sphinx.

**Comment by Cadair** [comment]
I am generally ok with this, I think a little more documentation about why these tools are useful would not go amiss. I think I understand from having a bit of knowledge about the cubes, but others may not.

**Comment by dpshelio** [comment]
@mateoi we need you to rebase this, thanks :)
Also, if you could make it python3 compatible would be great.  Just a couple of things to do with iterators here and there.

**Comment by dpshelio** [comment]
We need this to make EISpy to run... :(
@mateoi

**Comment by Cadair** [comment]
Feel free to update and reopen
